features/dnsresolver/netty/src/main/java/org/opennms/netmgt/dnsresolver/netty/NettyDnsResolver.java

```diff
@@ -257,10 +265,25 @@ public DnsServerAddressStreamProvider getNameServerProvider() {
     }
 
     public static List<InetSocketAddress> toSocketAddresses(String commaSeparatedAddressesWithPorts) {
-        final String servers[] = commaSeparatedAddressesWithPorts.split(",");
+        final String[] servers = commaSeparatedAddressesWithPorts.split(",");
         return Arrays.stream(servers)
                 .map(s -> {
-                    String parts[] = s.split(":");
+                    final String[] parts;
+                    final Matcher matcher = IPV6_ADDR_PATTERN.matcher(s);
+                    if (matcher.matches()) {
+                        // We've got an IPv6 address
+                        final String addrStr = matcher.group(1);
+                        final String portStr = matcher.group(3);
+                        if (portStr != null) {
+                            parts = new String[]{addrStr, portStr};
+                        } else {
+                            parts = new String[]{addrStr};
+                        }
+                    } else {
+                        // We've got an IPv4 address or a hostname
+                        parts = s.split(":");
+                    }
+
```

Comment:
Looks god. But I'm not sure if we can just split at `,` and use `HostAndPort` from guava for the parsing?

Reply:
Didn't know about that one, I'll try and use that instead.

Reply:
Done. Much simpler now.